app.js

```diff
@@ -1230,6 +1256,100 @@ define(function(require) {
 					err && err(parsedError);
 				}
 			});
+		},
+
+		updateUser: function(userData, callback) {
+			var self = this;
+
+			self.callApi({
+				resource: 'user.update',
+				data: {
+					accountId: self.accountId,
+					userId: userData.id,
+					data: userData
+				},
+				success: function(userData) {
+					callback && callback(userData);
+				}
+			});
+		},
+
+		listAccountDirectories: function(callback) {
+			var self = this;
+
+			self.callApi({
+				resource: 'directory.list',
+				data: {
+					accountId: self.accountId,
+					filters: {
+						paginate: 'false'
+					}
+				},
+				success: function(data) {
+					callback && callback(data.data);
+				}
+			});
+		},
+
+		usersCreateMainDirectory: function(callback) {
+			var self = this,
+				dataDirectory = {
+					confirm_match: false,
+					max_dtmf: 0,
+					min_dtmf: 3,
+					name: 'SmartPBX Directory',
+					sort_by: 'last_name'
+				};
+
+			self.callApi({
+				resource: 'directory.create',
+				data: {
+					accountId: self.accountId,
+					data: dataDirectory
+				},
+				success: function(data) {
+					callback && callback(data.data);
+				}
+			});
+		},
+
+		getMainDirectory: function(callback) {
+			var self = this;
+
+			self.listAccountDirectories(function(listDirectories) {
+				var indexMain = -1;
+
+				_.each(listDirectories, function(directory, index) {
+					if (directory.name === 'SmartPBX Directory') {
+						indexMain = index;
+
+						return false;
+					}
+				});
+
+				if (indexMain === -1) {
+					self.usersCreateMainDirectory(function(data) {
+						callback(data);
+					});
+				} else {
+					callback(listDirectories[indexMain]);
+				}
```

Comment:
This looks c/ped from some place else, but how about making it more declarative?

```js
var mainDirectory = _.find(listDirectories, { name: 'SmartPBX Directory' });

if (mainDirectory) {
    // mainDirectory exists
} else {
    // mainDirectory does not exists
}
```

Reply:
Fixed. Thanks :)